### ISSUE TITLE
Likely false warnings from switch fall-through lines

### ISSUE THREAD

**Issue by md0u80c9** [jimhester/lintr#334]
Hi,

If you use the fall-through syntax for the switch command, such as in the example below, you trigger warnings in lintr which I don't think are resolvable:

```
a <- "dog"

# Check how many legs
switch(a,
  "bird" = {
    # Two legs
  },
  "cat" = ,
  "dog" = ,
  "horse" = {
    # Four legs
  }
)
```

In the line `"cat" = ,` if you place a space between the = and the , you will get the warning:

`Commas should never have a space before.
`
If you don't, you'll get the warning:

`Put spaces around all infix operators.`

I suspect that this is a special case that `= ,` as a sequence is excepted (unless I'm missing something).

**Comment by pakuipers** [comment]
I was going to open an issue on this as well and agree that this should be addressed.

**Comment by MichaelChirico** [comment]
I think the bug is in `commas_linter`, as the Tidyverse style stance is to allow `"cat" = ,`, per https://github.com/tidyverse/style/issues/143

**Comment by MichaelChirico** [comment]
That said, I don't reproduce on current master:

# setup
```
writeLines('a <- "dog"

# Check how many legs
switch(a,
  "bird" = {
    # Two legs
  },
  "cat" = ,
  "dog" = ,
  "horse" = {
    # Four legs
  }
)', tmp<-tempfile())
```

# master
```
lintr::lint(tmp)
# <nothing>
```

# v2.0.1
```
lintr::lint(tmp)
/tmp/RtmppyRCut/file37ba7fa852aa:8:11: style: Commas should never have a space before.
  "cat" = ,
         ~^
/tmp/RtmppyRCut/file37ba7fa852aa:9:11: style: Commas should never have a space before.
  "dog" = ,
         ~^
```

**Comment by MichaelChirico** [comment]
Closed in #499
